vmdb/db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb

```diff
@@ -0,0 +1,48 @@
+#
+# Description: This method checks to see if the stack has been provisioned
+#   and whether the refresh has completed
+#
+
+def refresh_provider(service)
+  provider = service.orchestration_manager
+
+  $evm.log("info", "Refreshing provider #{provider.name}")
+  provider.refresh
+end
+
+$evm.log("info", "Check orchestration provisioned")
+
+service = $evm.root["service_template_provision_task"].destination
+
+if $evm.state_var_exist? 'stack_deployed'
+  # check whether refresh completed
+  stack = $evm.vmdb('orchestration_stack').find_by_ems_ref(service.stack_ems_ref)
+
+  $evm.log("info", "Check refresh status of name(#{stack.name}), stack(#{stack.inspect})")
```

Comment:
The stack variable can be nil so `stack.name` will cause an error.  Maybe move this down under the `if stack` block or get the expected stack_name from the service object.

Also stack.inspect seems like it might be logging too much information here.  Not sure it is really needed.